server/src/main/java/org/elasticsearch/index/query/RangeQueryBuilder.java

```diff
@@ -436,15 +437,17 @@ public String getWriteableName() {
     protected MappedFieldType.Relation getRelation(final CoordinatorRewriteContext coordinatorRewriteContext) {
         final MappedFieldType fieldType = coordinatorRewriteContext.getFieldType(fieldName);
         if (fieldType instanceof final DateFieldMapper.DateFieldType dateFieldType) {
-            if (coordinatorRewriteContext.hasTimestampData() == false) {
+            final IndexLongFieldRange fieldRange = coordinatorRewriteContext.getFieldRange(fieldName);
+            if (fieldRange.isComplete() == false || fieldRange == IndexLongFieldRange.EMPTY) {
+                // if not all shards for this (frozen) index have reported ranges to cluster state, OR if they
```

Comment:
Could they not report later for shards that just started? Shouldn't we err on the side of executing a query rather than not executing?